resilience4j-spring-boot-common/src/main/java/io/github/resilience4j/circuitbreaker/monitoring/health/CircuitBreakersHealthIndicator.java

```diff
@@ -37,20 +41,35 @@
     private static final String SLOW_CALLS = "slowCalls";
     private static final String NOT_PERMITTED = "notPermittedCalls";
     private static final String STATE = "state";
-    private final CircuitBreaker circuitBreaker;
 
-    public CircuitBreakerHealthIndicator(CircuitBreaker circuitBreaker) {
-        this.circuitBreaker = circuitBreaker;
+    private final CircuitBreakerRegistry circuitBreakerRegistry;
+    private final CircuitBreakerConfigurationProperties circuitBreakerProperties;
+    private final HealthAggregator healthAggregator;
+
+    public CircuitBreakersHealthIndicator(CircuitBreakerRegistry circuitBreakerRegistry,
+                                          CircuitBreakerConfigurationProperties circuitBreakerProperties,
+                                          HealthAggregator healthAggregator) {
+        this.circuitBreakerRegistry = circuitBreakerRegistry;
+        this.circuitBreakerProperties = circuitBreakerProperties;
+        this.healthAggregator = healthAggregator;
     }
 
     @Override
     public Health health() {
-        return Optional.of(circuitBreaker)
-                .map(this::mapBackendMonitorState)
-                .orElse(Health.up().build());
+        Map<String, Health> healths = circuitBreakerRegistry.getAllCircuitBreakers().toJavaStream()
+                .filter(this::isRegisterHealthIndicator)
+                .collect(Collectors.toMap(CircuitBreaker::getName, CircuitBreakersHealthIndicator::mapBackendMonitorState));
+
+        return healthAggregator.aggregate(healths);
+    }
+
+    private boolean isRegisterHealthIndicator(CircuitBreaker circuitBreaker) {
```

Comment:
This method is used to check `registerHealthIndicator` value. It is same with `RateLimitersHealthIndicator`. 
In this method, `circuitBreakerProperties.findCircuitBreakerProperties(name)` always null  with **dynamic instance** because dynamic instance doesn't have any InstanceProperties. 
So **dynamic instance always register healthIndicator**. Is it ok?

Reply:
Yes, that should be ok.